:warning: Action Required: Replace Deprecated gcr.io/kubebuilder/kube-rbac-proxy

## Description

:warning: **The image `gcr.io/kubebuilder/kube-rbac-proxy` is deprecated and will become unavailable.**
**You must move as soon as possible, sometime from early 2025, the GCR will go away.**

> _Unfortunately, we're unable to provide any guarantees regarding timelines or potential extensions at this time. Images provided under GRC will be unavailable from March 18, 2025, [as per announcement](https://cloud.google.com/artifact-registry/docs/transition/transition-from-gcr). However, `gcr.io/kubebuilder/`may be unavailable before this date due [to efforts to deprecate infrastructure](https://github.com/kubernetes/k8s.io/issues/2647)._

- **If your project uses `gcr.io/kubebuilder/kube-rbac-proxy`, it will be affected.**
  Your project may fail to work if the image cannot be pulled. **You must take action as soon as possible.**

- However, if your project is **no longer using this image**, **no action is required**, and you can close this issue.

## Using the image `gcr.io/kubebuilder/kube-rbac-proxy`?

[kube-rbac-proxy](https://github.com/brancz/kube-rbac-proxy) was historically used to protect the metrics endpoint. However, its usage has been discontinued in [Kubebuilder](https://github.com/kubernetes-sigs/kubebuilder). The default scaffold now leverages the [`WithAuthenticationAndAuthorization`](https://pkg.go.dev/sigs.k8s.io/controller-runtime/pkg/metrics/filters#WithAuthenticationAndAuthorization) feature provided by [Controller-Runtime](https://github.com/kubernetes-sigs/controller-runtime).

This feature provides integrated support for securing metrics endpoints by embedding authentication (`authn`) and authorization (`authz`) mechanisms directly into the controller manager's metrics server, replacing the need for (https://github.com/brancz/kube-rbac-proxy) to secure metrics endpoints.

### What To Do?

**You must replace the deprecated image `gcr.io/kubebuilder/kube-rbac-proxy` with an alternative approach. For example:**
- Update your project to use [`WithAuthenticationAndAuthorization`](https://pkg.go.dev/sigs.k8s.io/controller-runtime/pkg/metrics/filters#WithAuthenticationAndAuthorization):
  > _You can fully upgrade your project to use the latest scaffolds provided by the tool or manually make the necessary changes. Refer to the [FAQ and Discussion](https://github.com/kubernetes-sigs/kubebuilder/discussions/3907) for detailed instructions on how to manually update your project and test the changes._
- Alternatively, replace the image with another trusted source at your own risk, as its usage has been discontinued in Kubebuilder.

**For further information, suggestions, and guidance:**
- 📖 [FAQ and Discussion](https://github.com/kubernetes-sigs/kubebuilder/discussions/3907)
- 💬 Join the Slack channel: [#kubebuilder](https://communityinviter.com/apps/kubernetes/community#kubebuilder).

> **NOTE:** _This issue was opened automatically as part of our efforts to identify projects that might be affected and to raise awareness about this change within the community. If your project is no longer using this image, **feel free to close this issue**._

We sincerely apologize for any inconvenience this may cause.

**Thank you for your cooperation and understanding!** :pray:

Raised a PR but not sure how to test this locally. Tried running `make install deploy` but getting the following:
```
anshuman@Anshumans-MacBook-Pro lws % make install deploy
test -s /Users/anshuman/work/lws/bin/controller-gen && /Users/anshuman/work/lws/bin/controller-gen --version | grep -q v0.16.2 || \
        GOBIN=/Users/anshuman/work/lws/bin go install sigs.k8s.io/controller-tools/cmd/controller-gen@v0.16.2
/Users/anshuman/work/lws/bin/controller-gen \
                rbac:roleName=manager-role output:rbac:artifacts:config=config/rbac \
                crd:generateEmbeddedObjectMeta=true output:crd:artifacts:config=config/crd/bases \
                webhook output:webhook:artifacts:config=config/webhook \
                paths="./..."
/Users/anshuman/work/lws/bin/kustomize build config/crd | kubectl create -f -
error: no objects passed to create
make: *** [install] Error 1
```

--- a/charts/lws/README.md
+++ b/charts/lws/README.md
@@ -52,9 +52,6 @@ The following table lists the configurable parameters of the LWS chart and their
 | `image.manager.repository`                  | Repository for manager image                   | `gcr.io/k8s-staging-lws/lws`         |
 | `image.manager.tag`                         | Tag for manager image                          | `main`                               |
 | `image.manager.pullPolicy`                  | Pull policy for manager image                  | `IfNotPresent`                       |
-| `image.kubeRbacProxy.repository`            | Repository for kubeRbacProxy image             | `gcr.io/kubebuilder/kube-rbac-proxy` |
-| `image.kubeRbacProxy.tag`                   | Tag for kubeRbacProxy image                    | `v0.15.0`                            |
-| `image.kubeRbacProxy.pullPolicy`            | Pull policy for kubeRbacProxy image            | `IfNotPresent`                       |
 | `podAnnotations`                            | Annotations for pods                           | `{}`                                 |
 | `podSecurityContext.runAsNonRoot`           | Run pod as non-root user                       | `true`                               |
 | `securityContext.allowPrivilegeEscalation`  | Allow privilege escalation in security context | `false`                              |

--- a/charts/lws/values.yaml
+++ b/charts/lws/values.yaml
@@ -18,12 +18,6 @@ image:
     # tag, if defined will use the given image tag, else Chart.AppVersion will be used
     tag: main
     pullPolicy: IfNotPresent
-  kubeRbacProxy:
-    repository: gcr.io/kubebuilder/kube-rbac-proxy
-    # tag
-    tag: v0.15.0
-    # This should be set to 'IfNotPresent' for released version
-    pullPolicy: IfNotPresent
 
 podAnnotations: {}
 

